tests/test_content_client.py

```diff
@@ -0,0 +1,170 @@
+import pytest
+import testinfra
+import time
+import subprocess
+
+
+@pytest.fixture
+def katello_client(request, entities, registration_hostname):
+    image_id = subprocess.check_output(['podman', 'images', '-q', '--filter', f"reference={request.param}"])
+    assert image_id
+
+    container_id = subprocess.check_output([
+        'podman',
+        'run',
+        '--volume',
+        '/dev/log:/dev/log',
+        '--detach',
+        '--tty',
+        request.param
+    ]).decode().strip()
+
+    try:
+        container = testinfra.get_host("podman://" + container_id)
+
+        container.run_expect([0], f"rpm -Uvh http://{registration_hostname}/pub/katello-ca-consumer-latest.noarch.rpm")
+        container.run_expect([0], f"subscription-manager register --force --org={entities['organization_label']} --activationkey=\"{entities['activation_key']}\"")
+
+        consumed = container.check_output("subscription-manager list --consumed")
+        assert entities['product'] in consumed
+
+        # return a testinfra connection to the container
+        yield container
+
+    finally:
+        subprocess.check_call(['podman', 'rm', '-f', container_id])
+
+
+@pytest.fixture
+def katello_agent_client(katello_client, os_release_major):
+    if os_release_major not in ("6", "7"):
+        pytest.skip(f"katello-agent not available on EL{os_release_major}")
```

Comment:
Is this still true? It's now gotten more complicated.

Reply:
I think this is true enough for most katello releases to be kept as is and we can add the "el8&&k4+" logic at some point later.